composables/api/useUserSignup.ts

```diff
@@ -0,0 +1,39 @@
+import { useMutation } from 'vue-query';
+import type { Database } from '~/types/generated-types';
+
+export const useUserSignup = () => {
+  const client = useSupabaseClient<Database>();
+
+  const result = useMutation(
+    async (credentials: { email: string; password: string; name: string }) => {
+      const { email, password, name } = credentials;
+
+      const {
+        data: { user },
+        error,
+      } = await client.auth.signUp({ email, password });
+
+      if (error) {
+        console.error(error);
+        throw new Error(error.message);
+      }
+
+      const userProfileData = {
+        id: user?.id,
+        username: name,
+      };
+
+      const { error: dbError } = await client
+        .from('test_profiles')
+        .insert(userProfileData);
+
+      if (dbError) {
+        console.error(dbError);
+        throw new Error(dbError.message);
+      }
```

Comment:
**Important point**

I added creating user profile in database after user registration. I believe user and user-profile are different things regarding backend logic. `users` table in our database is in the `auth` schema, which is being protected and we can't really add or remove columns to it. Also, this way removing a user doesn't remove this user's profile.

I'm not familiar with the rules of this database, so `test_profiles` is now a table of user profile data in the `public` schema. It would be good of course to change it to `profiles` table in a schema that is not being available to everyone, but for now I don't know how to do it.

Reply:
It's normal that auth-users is not the same as profile-users, but they have 1-1 relation between, and can be strictly binded to each other.

About test profiles may be it's because of that you're running DB locally? 🤔

Reply:
I just didn't set up anything and blindly added empty 'profiles' table to 'public' schema, so I can register users. I need help with how to set up the security there, create a proper schema for such table etc.